### ISSUE TITLE
Feature Proposal :  CAN Protocol Adapter (Provider) with SocketCAN and ISO-TP support for request-response capabilities.

### ISSUE THREAD

**Issue by RaviprasadBS** [eclipse-kuksa/kuksa-incubation#14]
**Background**

The current  **Kuksa-can-provider** only supports reading CAN frames from the bus and lacks functionality for sending requests over the CAN bus, except for actuator signals. This limitation restricts use cases for applications that require two-way vehicle data exchange, such as diagnostics, monitoring, and control.

**Proposed Solution** 

The proposed solution aims to enhance CAN communication by integrating Socket CAN and ISO-TP support for both request and response operations, addressing the current limitations in two-way data exchange. This solution enhances the ability to interact with CAN networks for diagnostics, monitoring, and control. By utilizing DBC support, CAN data can be decoded and the actual values can be fed into Kuksa data broker.

**Key Features** 

**User Configuration :** JSON-based user configuration for setting up CAN and Socket CAN interfaces, defining CAN request-response tables, specifying protocol parameters, and mapping VSS and PID signal names.

**Socket CAN Integration** : Compatible with Socket CAN interfaces for CAN bus communication in Linux-based environments.

**Read CAN Frame and Decode:** Read CAN frames from the bus and decode them using DBC files, mapping the signals to their corresponding values.

**Register and Feed Data Points to Data Broker:** Register VSS signal data points with Kuksa Data Broker and feed the decoded or interpreted values into the data broker.

**Technical context**
![Technical context](https://github.com/user-attachments/assets/15d8c998-dbf0-4a3c-b1ce-bb2850165ae7)

**Programming language and Dependencies**

- Programming Language : **Rust**
- Linux Dependencies : **socket-can and can-isotp driver.**

**Expected Outcome** 

- A Rust-based CAN Protocol Adapter capable of configuring the CAN bus, sending user-defined CAN requests, receiving responses on a CAN network, and feeding the decoded values as VSS signals to the Kuksa DataBroker.

- Basic documentation, including usage examples and integration guide with Kuksa data broker and linux systems.

**Comment by lukasmittag** [comment]
#18 merged so done? @RaviprasadBS can we close this?

**Comment by RaviprasadBS** [comment]
> [#18](https://github.com/eclipse-kuksa/kuksa-incubation/pull/18) merged so done? [@RaviprasadBS](https://github.com/RaviprasadBS) can we close this?

Yes https://github.com/eclipse-kuksa/kuksa-incubation/pull/18 is merged. We can close this ticket.
Thanks.

**Comment by lukasmittag** [comment]
close as done
